File: golang/codec/message_windowadjust.go
package codec

import (
	"encoding/binary"
	"fmt"
)

type WindowAdjustMessage struct {
	ChannelID       uint32
	AdditionalBytes uint32
}

func (msg WindowAdjustMessage) String() string {
	return fmt.Sprintf("{WindowAdjustMessage ChannelID:%d AdditionalBytes:%d}",
		msg.ChannelID, msg.AdditionalBytes)
}

func (msg WindowAdjustMessage) Channel() (uint32, bool) {
	return msg.ChannelID, true
}

func (msg WindowAdjustMessage) MarshalMux() ([]byte, error) {
	packet := make([]byte, payloadSizes[msgChannelWindowAdjust]+1)
	packet[0] = msgChannelWindowAdjust
	binary.BigEndian.PutUint32(packet[1:5], msg.ChannelID)
	binary.BigEndian.PutUint32(packet[5:9], msg.AdditionalBytes)
	return packet, nil
}

func (msg *WindowAdjustMessage) UnmarshalMux(b []byte) error {
	msg.ChannelID = binary.BigEndian.Uint32(b[1:5])
	msg.AdditionalBytes = binary.BigEndian.Uint32(b[5:9])
	return nil
}

File: csharp/qmux/codec/message_windowadjust.cs
namespace qmux.codec;

using System;
using System.IO;
using System.Text;

public struct WindowAdjustMessage
{
    public UInt32 ChannelId;
    public UInt32 AdditionalBytes;
    public string String()
    {
        return $"OpenConfirmMessage ChannelId:{this.ChannelId} AdditionalBytes:{this.AdditionalBytes}";
    }
    public (UInt32, bool) Channel()
    {
        return (this.ChannelId, true);
    }

    public byte[] MarshalMux()
    {
        // 8 + 1
        var bufferSize = (int)PayloadSizes.MessageChannelWindowAdjust + 1;
        var stream = new MemoryStream(new byte[bufferSize]);
        using (var writer = new BinaryWriter(stream, Encoding.BigEndianUnicode, false))
        {
            // 0
            writer.Write((byte)MessageType.MessageChannelWindowAdjust);
            // 1-5
            writer.Write(this.ChannelId);
            writer.Write(this.AdditionalBytes);
        }
        return stream.ToArray();
    }
    public void UnmarshalMux(byte[] b)
    {
        var stream = new MemoryStream(b);
        using (var reader = new BinaryReader(stream, Encoding.BigEndianUnicode))
        {
            // 0
            reader.ReadByte();
            // 1-5
            this.ChannelId = reader.ReadUInt32();
            this.AdditionalBytes = reader.ReadUInt32();
        }
    }
}